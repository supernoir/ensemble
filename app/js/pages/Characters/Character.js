import React from 'react';
import axios from 'axios';
import intl from 'react-intl-universal';
import { Link } from 'react-router-dom';
import { Container, Breadcrumb, Segment, Divider, Header, Button, Icon } from 'semantic-ui-react';

export default class Character extends React.Component {
	constructor(){
		super();
		this.state = {
			character: []
		};
	}
	componentDidMount(){
		axios.get(`http://localhost:3030/character/${this.props.match.params.id}`)
			.then(res => {
				this.setState({ character: res.data.character });
			}
			);
	}

	render(){
		return(
			<Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section link>
						<Link to="/characters">{intl.get('entity.characters')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						{this.state.character.first_name} {this.state.character.last_name}
					</Breadcrumb.Section>
				</Breadcrumb>
				<Segment>
					<Header as='h2'>
						{this.state.character.first_name} <b>{this.state.character.last_name}</b>
						<Header.Subheader>{this.state.character.gender}, {this.state.character.age}</Header.Subheader>
					</Header>
					<Divider/>

					<p>
						<Icon name='info circle'/>{this.state.character.desc}
					</p>

					<Divider/>
					<Button>
						{intl.get('character.action-edit')}
					</Button>
					<Button>
						{intl.get('character.action-delete')}
					</Button>
				</Segment>

			</Container>
		);
	}
}