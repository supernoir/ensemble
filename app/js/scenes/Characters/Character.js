import React from 'react';
import intl from 'react-intl-universal';
import { Link } from 'react-router-dom';
import { Container, Breadcrumb, Segment, Divider, Header, Button, Icon } from 'semantic-ui-react';
import Loader from '../../basics/Loader';

export default class Character extends React.Component {
	constructor(){
		super();
		this.state = {
			character: []
		};
	}
	componentDidMount(){
		this.props.getCharacterById(this.props.match.params.id);
	}

	render(){
		return this.props.loading
			? <Loader loading={this.props.loading} />
			:	<Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section link>
						<Link to="/characters">{intl.get('entity.characters')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						{this.props.character.first_name} {this.props.character.last_name}
					</Breadcrumb.Section>
				</Breadcrumb>
				<Segment>
					<Header as='h2'>
						{this.props.character.first_name} <b>{this.props.character.last_name}</b>
						<Header.Subheader>{this.props.character.gender}, {this.props.character.age}</Header.Subheader>
					</Header>
					<Divider/>

					<p>
						<Icon name='info circle'/>{this.props.character.desc}
					</p>

					<Divider/>
					<Button>
						{intl.get('character.action-edit')}
					</Button>
					<Button>
						{intl.get('character.action-delete')}
					</Button>
				</Segment>
			</Container>;
	}
}