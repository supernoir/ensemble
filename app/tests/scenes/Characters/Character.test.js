import React from 'react';
import { shallow } from 'enzyme';

// Import Mocks
import mockProject from '../../mocks/mockProject';
import mockCharacter from '../../mocks/mockCharacter';

// Import Components
import Character from '../../../js/scenes/Characters/Character';

test('Project -- Snapshot Test', () => {
	const wrapper = shallow(
		<Character
			loading={false}
			getCharacterById={id => jest.fn(id)}
			project={mockProject}
			character={mockCharacter}
			deleteSpecificCharacter={id => jest.fn(id)}
			match={{ params: { id: 1 } }}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});
