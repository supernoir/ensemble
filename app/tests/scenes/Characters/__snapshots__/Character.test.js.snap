// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project -- Snapshot Test 1`] = `
<Container>
  <Breadcrumb>
    <BreadcrumbSection>
      <Link
        replace={false}
        to="/"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection>
      <Link
        replace={false}
        to="/characters"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection
      active={true}
    >
      Michelle Romaine Toulouse-Lautrec
    </BreadcrumbSection>
  </Breadcrumb>
  <SegmentGroup>
    <Segment>
      <Header
        as="h2"
      >
        Michelle Romaine Toulouse-Lautrec
      </Header>
    </Segment>
    <SegmentGroup
      horizontal={true}
    >
      <Segment>
        <Header
          as="h4"
        />
        <Label
          content="Non-binary"
          icon="transgender"
          size="large"
        />
      </Segment>
      <Segment>
        <Header
          as="h4"
        />
        <Label
          content="2015-03-03"
          icon="birthday cake"
          size="large"
        />
      </Segment>
    </SegmentGroup>
    <Segment>
      <Header
        as="h4"
      />
      <p>
        Un ecrivain
      </p>
    </Segment>
    <Segment>
      <Header
        as="h4"
      />
      <p>
        Paris, France
      </p>
    </Segment>
    <Segment>
      <Header
        as="h4"
      />
      <Label
        content="5d17dafd4ea603c5dcff98de"
        icon="book"
        key="5d17dafd4ea603c5dcff98de-0"
        size="large"
      />
      <Label
        content="5d13a6864ea603c5dcff98d8"
        icon="book"
        key="5d13a6864ea603c5dcff98d8-1"
        size="large"
      />
    </Segment>
    <Segment>
      <Link
        replace={false}
        to="/editcharacter/5d57d0d2c39bc90249b6917c"
      >
        <Button
          as="button"
        />
      </Link>
      <Button
        as="button"
        content=""
        onClick={[Function]}
      />
    </Segment>
  </SegmentGroup>
</Container>
`;
